osbenchmark/builder/downloaders/builders/source_builder.py

```diff
@@ -0,0 +1,40 @@
+import logging
+import os
+
+from osbenchmark.builder.utils.jdk_resolver import JdkResolver
+from osbenchmark.builder.utils.path_manager import PathManager
+from osbenchmark.exceptions import ExecutorError, BuildError
+
+
+class SourceBuilder:
+    def __init__(self, executor, source_directory, build_jdk, log_directory):
+        self.logger = logging.getLogger(__name__)
+        self.executor = executor
+        self.path_manager = PathManager(executor)
+        self.jdk_resolver = JdkResolver(executor)
+
+        self.source_directory = source_directory
+        self.build_jdk = build_jdk
+        self.log_directory = log_directory
+
+    def build(self, host, build_commands, override_source_directory=None):
+        if isinstance(build_commands, str):
```

Comment:
Is there only going to be this Builder? I'm wondering if the type check makes sense to include if this is using an interface that multiple builders would implement.

Minor: But if this is the only Builder then line 24 in `CorePluginSourceDownloader` could just pass the string instead of a list. I would also consider changing the name of the field in that class to be `source_builder instead` of just `builder`

Reply:
This is the only one as of now. There could very well be a need for other ones in the future so I'll refactor it into an interface. I'll get rid of the typecheck then too, was added as I made the mistake of passing a string in the unit tests and figured that could be an easy mistake for others to make as well :)